Modules/_pickle.c

```diff
@@ -1168,10 +1168,13 @@ _Pickler_New(PickleState *st)
     self->reducer_override = NULL;
 
     self->memo = PyMemoTable_New();
+    if (self->memo == NULL) {
+        Py_DECREF(self);
```

Comment:
It is not safe to call `Py_DECREF(self)` here, because `self->output_buffer` is not initialized yet.

Reply:
`PyObject_GC_New` returns a zeroed context, no?

Reply:
Not sure. I looked at the code and have not find where it can zero content.

Reply:
`PyObject_GC_New` is documented as “Analogous to [PyObject_New()](https://docs.python.org/3/c-api/allocation.html#c.PyObject_New)”, whose docs explicitly say “Fields not defined by the Python object header are not initialized”

`PyType_GenericAlloc` -- which most types put in `tp_alloc` -- is what zeroes memory for a new object. AFAIK, direct calls to `_Pickler_New` bypass that.

Reply:
Thanks, Petr!

Reply:
See gh-105686 for the amendment.